ale/driver_verification.py

```diff
@@ -0,0 +1,217 @@
+    instrument_pointing['ephemeris_times'] = time_dependent_rotation.times
+    instrument_pointing['quaternions'] = time_dependent_rotation.quats[:, [3, 0, 1, 2]]
+    instrument_pointing['angular_velocities'] = time_dependent_rotation.av
+
+    # reference frame should be the last frame in the chain
+    instrument_pointing["reference_frame"] = instrument_pointing['time_dependent_frames'][-1]
+
+    # Reverse the frame order because ISIS orders frames as
+    # (destination, intermediate, ..., intermediate, source)
+    instrument_pointing['constant_frames'] = shortest_path(frame_chain, sensor_frame, destination_frame)
+    constant_rotation = frame_chain.compute_rotation(destination_frame, sensor_frame)
+    instrument_pointing['constant_rotation'] = constant_rotation.rotation_matrix().flatten()
+    
+    return instrument_pointing
+
+def generate_instrument_position(driver):
+    instrument_position = {}
+    positions, velocities, times = driver.sensor_position
+    instrument_position['spk_table_start_time'] = times[0]
+    instrument_position['spk_table_end_time'] = times[-1]
+    instrument_position['spk_table_original_size'] = len(times)
+    instrument_position['ephemeris_times'] = times
+    # Rotate positions and velocities into J2000 then scale into kilometers
+    # velocities = j2000_rotation.rotate_velocity_at(positions, velocities, times)/1000
+    # positions = j2000_rotation.apply_at(positions, times)/1000
+    instrument_position['positions'] = positions
+    instrument_position['velocities'] = velocities
+    return instrument_position
+
+def generate_sun_position(driver):
+    sun_position = {}
+    positions, velocities, times = driver.sun_position
+    sun_position['spk_table_start_time'] = times[0]
+    sun_position['spk_table_end_time'] = times[-1]
+    sun_position['spk_table_original_size'] = len(times)
+    sun_position['ephemeris_times'] = times
+    # Rotate positions and velocities into J2000 then scale into kilometers
+    # velocities = j2000_rotation.rotate_velocity_at(positions, velocities, times)/1000
+    # positions = j2000_rotation.apply_at(positions, times)/1000
+    sun_position['positions'] = positions
+    sun_position['velocities'] = velocities
+    # sun_position["reference_frame"] = j2000_rotation.dest
+    return sun_position
+
+def create_json_dump(driver, sensor_frame_id, target_frame_id):
+    json_dump = {}
+    json_dump["instrument_rotation"] = generate_instrument_rotation(driver, sensor_frame_id)
+    json_dump["body_rotation"] = generate_body_rotation(driver, target_frame_id)
+    json_dump["instrument_position"] = generate_instrument_position(driver)
+    json_dump["sun_position"] = generate_sun_position(driver)
+    return json_dump
+
+def diff_and_describe(json1, json2, key_array):
+    for key in key_array:
+        json1 = json1[key]
+        json2 = json2[key]
+    diff = json1 - json2
+    print(len(diff), np.mean(diff, axis=(0)), np.median(diff, axis=(0)))
```

Comment:
Can we add a little to this print statement?

Basically change it to something like:
`print(" ".join(key_array) + "\nNum records:", len(diff), "\nMean:", np.mean(diff, axis=(0)), "\nMedian:", np.median(diff, axis=(0)), "\n")`

It should help to clarify what is being compared and what the numbers/arrays are

Reply:
updated in latest commit!